metricbeat/module/prometheus/remote_write/data.go

```diff
@@ -57,13 +57,16 @@ func (p *remoteWriteEventGenerator) GenerateEvents(metrics model.Samples) map[st
 			labels[string(k)] = v
 		}
 
-		// join metrics with same labels in a single event
-		labelsHash := labels.String()
+		// join metrics with same labels and same timestamp in a single event
+		timeStampedLabels := labels.Clone()
+		timeStampedLabels.Put("timestamp", metric.Timestamp.Time())
+		labelsHash := timeStampedLabels.String()
```

Comment:
I would try to avoid cloning to save memory and cycles. How about appending the timestamp to the labelsHash?

Something like `sampleHash := labels.String() + metric.Timestamp.Time()`

Reply:
👍